libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -44,14 +44,69 @@ const std::string kMinimalUnknownVersionShader =
     "#version 550\n"
     "void main() {}\n";
 
-TEST(CppInterface, MultipleCalls) {
+class CppInterface : public testing::Test {
+ protected:
+  // Compiles a shader and returns true on success, false on failure.
+  bool CompilationSuccess(const std::string& shader,
+                          shaderc_shader_kind kind) const {
+    return compiler_.CompileGlslToSpv(shader.c_str(), shader.length(), kind)
+        .GetSuccess();
+  }
+
+  // Compiles a shader with options and returns true on success, false on
+  // failure.
+  bool CompilationSuccess(const std::string& shader, shaderc_shader_kind kind,
+                          const shaderc::CompileOptions& options) const {
+    return compiler_.CompileGlslToSpv(shader.c_str(), shader.length(), kind,
+                                      options)
+        .GetSuccess();
+  }
+
+  // Compiles a shader and returns true if the result is valid SPIR-V.
+  bool CompilesToValidSpv(const std::string& shader,
+                          shaderc_shader_kind kind) const {
+    return IsValidSpv(compiler_.CompileGlslToSpv(shader, kind));
+  }
+
+  // Compiles a shader with options and returns true if the result is valid
+  // SPIR-V.
+  bool CompilesToValidSpv(const std::string& shader, shaderc_shader_kind kind,
+                          const shaderc::CompileOptions& options) const {
+    return IsValidSpv(compiler_.CompileGlslToSpv(shader, kind, options));
+  }
+
+  // Compiles a shader, asserts compilation success, and returns the error
+  // message.
+  std::string CompilationErrors(const std::string& shader,
+                                shaderc_shader_kind kind,
+                                const shaderc::CompileOptions& options) {
+    const auto module = compiler_.CompileGlslToSpv(shader, kind, options);
+    EXPECT_TRUE(module.GetSuccess()) << kind << '\n' << shader;
```

Comment:
I think there are cases that the compilation can fail and we want to check the specific error message.

E.g. when an error is caused by a wrong version-profile, we need to make sure the compilation fail and the error message shows the wrong version-profile.

I think we don't need to assert compilation success here.